src/Binding/JSSystem.cpp

```diff
@@ -118,15 +91,35 @@ static bool nidium_nidium_language(JSContext *cx, unsigned argc, jsval *vp)
 }
 #endif
 #endif
-// }}}
 
-// {{{ Registration
+#if 0
+JSPropertySpec *JSSystem::ListProperties()
+{
+    static JSPropertySpec props[] = {
+#ifdef NIDIUM_PRODUCT_UI
+        CLASSMAPPER_PROP_G(JSSystem, language),
+#endif
+
+        JS_PS_END
+    };
+    return props;
+}
+#endif
+
+JSFunctionSpec *JSSystem::ListMethods()
+{
+    static JSFunctionSpec funcs[] = {
+        CLASSMAPPER_FN(JSSystem, getOpenFileStats, 0),
+        JS_FS_END
+    };
+
+    return funcs;
+}
+
 void JSSystem::RegisterObject(JSContext *cx)
 {
-    JS::RootedObject global(cx, JS::CurrentGlobalOrNull(cx));
-    JS::RootedObject systemObj(
-        cx, JS_DefineObject(cx, global, "System", &system_class, nullptr, 0));
-    JS_DefineFunctions(cx, systemObj, system_funcs);
+    JSSystem::ExposeClass<0>(cx, "System");
```

Comment:
The className needs to be different. You can set this to "NidiumSystem".
Also the <0> is not required